1-Draft/RFCNNNN-Timers.md

```diff
@@ -0,0 +1,44 @@
+---
+RFC: RFCNNNN
+Author: Rob Pleau
+Status: Draft
+Version: 1.0
+Area: Microsoft.PowerShell.Utility / Commands
+Comments Due:
+Plan to implement: Yes
+---
+
+# Timer Functionality in PowerShell
+
+Many users of PowerShell have a need to create a timer.  Typically users doing this are doing so by manually instantiating an instance of `System.Diagnostics.Stopwatch`.
+
+## Motivation
+
+As a *PowerShell User*
+I want to *Create timers for automation jobs*
+So that *I can have logic stop a job that runs to long*
```

Comment:
I think this RFC is leaking implementation details and what is really needed is a single `Wait-Timeout` cmdlet that takes a `[ScriptBlock]` parameter. No one really wants to mess with timers for this scenario as that is more code to write.

Are there other common scenarios that would drive a need for `Timer` cmdlets? I wouldn't use this for Benchmarking because it would be its own set of abstractions to get right. I can't think of any others.

Reply:
@ephos I appreciate you taking the time to submit this, however, I agree that if the primary scenario is to stop execution after some timeout, I think https://github.com/PowerShell/PowerShell/issues/5434 already captures that.

Reply:
You guys make fair points.  I agree [#5434](https://github.com/PowerShell/PowerShell/issues/5434) probably would be a better solution for the scenarios I have had to use `[System.Diagnostics.StopWatch]` to work around.